### PR TITLE
baeldung-articles : BAEL-7658

### PR DESCRIPTION
**Pull request by DiegoMarti2**
Convert InputStream into Stream<String> in Java.

### CODE REVIEW

File: core-java-modules/core-java-streams-6/src/test/java/com/baeldung/streams/inputstreamintostringstream/InputStreamToStringStreamUnitTest.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Test
+    public void givenInputStream_whenConvertingWithScannerFindAll_thenConvertInputStreamToStringStream() {
+        try (Scanner scanner = new Scanner(inputStream, StandardCharsets.UTF_8)) {
+            Stream<String> stringStream = scanner.findAll(".+")
+                    .map(MatchResult::group);
```
Comment (theangrydev) [inline core-java-modules/core-java-streams-6/src/test/java/com/baeldung/streams/inputstreamintostringstream/InputStreamToStringStreamUnitTest.java:37]: This formatting looks like you are not using the Baeldung formatter:
https://github.com/Baeldung/dev-settings/tree/main/javaCodeStyle/intellij

**Review by theangrydev — CHANGES_REQUESTED** [review]

File: core-java-modules/core-java-streams-6/src/test/java/com/baeldung/streams/inputstreamintostringstream/InputStreamToStringStreamUnitTest.java (hunk 1 above)
Comment (DiegoMarti2) [reply core-java-modules/core-java-streams-6/src/test/java/com/baeldung/streams/inputstreamintostringstream/InputStreamToStringStreamUnitTest.java:35]: I am using this format.

Comment (theangrydev) [reply core-java-modules/core-java-streams-6/src/test/java/com/baeldung/streams/inputstreamintostringstream/InputStreamToStringStreamUnitTest.java:35]: Please check again, the guidelines are a 4 space indent not 8 space

Comment (DiegoMarti2) [reply core-java-modules/core-java-streams-6/src/test/java/com/baeldung/streams/inputstreamintostringstream/InputStreamToStringStreamUnitTest.java]: Fixed.

File: core-java-modules/core-java-streams-6/src/test/java/com/baeldung/streams/inputstreamintostringstream/InputStreamToStringStreamUnitTest.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
+package inputstreamintostringstream;
```
Comment (theangrydev) [inline core-java-modules/core-java-streams-6/src/test/java/com/baeldung/streams/inputstreamintostringstream/InputStreamToStringStreamUnitTest.java:1]: Package doesn't match the package path, also let's rename it as:
```suggestion
package com.baeldung.streams.inputstreamtostringstream;
```

File: core-java-modules/core-java-streams-6/src/test/java/com/baeldung/streams/inputstreamintostringstream/InputStreamToStringStreamUnitTest.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+import org.junit.jupiter.api.Test;
+
+import java.io.*;
+import java.nio.charset.StandardCharsets;
+import java.util.Scanner;
+import java.util.regex.MatchResult;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.junit.Assert.assertEquals;
```
Comment (theangrydev) [inline core-java-modules/core-java-streams-6/src/test/java/com/baeldung/streams/inputstreamintostringstream/InputStreamToStringStreamUnitTest.java:12]: Using the wrong assertEquals, import the JUnit5 one
```suggestion
import static org.junit.jupiter.api.Assertions.assertEquals;
```

File: core-java-modules/core-java-streams-6/src/test/java/com/baeldung/streams/inputstreamintostringstream/InputStreamToStringStreamUnitTest.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+            Stream<String> stringStream = reader.lines();
+
+            String result = stringStream.reduce("", (s1, s2) -> s1 + s2);
+
+            assertEquals("HelloWorldThisisatest", result);
+        }
+    }
+
+    @Test
+    void givenInputStream_whenConvertingWithScannerFindAll_thenConvertInputStreamToStringStream() throws IOException {
```
Comment (theangrydev) [inline core-java-modules/core-java-streams-6/src/test/java/com/baeldung/streams/inputstreamintostringstream/InputStreamToStringStreamUnitTest.java:32]: No exception is needed to be declared here
```suggestion
    void givenInputStream_whenConvertingWithScannerFindAll_thenConvertInputStreamToStringStream() {

```

File: core-java-modules/core-java-streams-6/src/test/java/com/baeldung/streams/inputstreamintostringstream/InputStreamToStringStreamUnitTest.java [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+import static org.junit.Assert.assertEquals;
+
+public class InputStreamToStringStreamUnitTest {
+
+    byte[] bytes = "Hello\nWorld\nThis\nis\na\ntest".getBytes(StandardCharsets.UTF_8);
+    InputStream inputStream = new ByteArrayInputStream(bytes);
+
+    @Test
+    void givenInputStream_whenConvertingWithBufferedReader_thenConvertInputStreamToStringStream() throws IOException {
+        try (InputStreamReader isr = new InputStreamReader(inputStream, StandardCharsets.UTF_8); BufferedReader reader = new BufferedReader(isr)) {
```
Comment (theangrydev) [inline core-java-modules/core-java-streams-6/src/test/java/com/baeldung/streams/inputstreamintostringstream/InputStreamToStringStreamUnitTest.java:21]: The formatting that was here previously was better and already correct:
```suggestion
        try (InputStreamReader isr = new InputStreamReader(inputStream, StandardCharsets.UTF_8);
             BufferedReader reader = new BufferedReader(isr)) {

```

**Review by theangrydev — CHANGES_REQUESTED** [review]

File: core-java-modules/core-java-streams-6/src/test/java/com/baeldung/streams/inputstreamintostringstream/InputStreamToStringStreamUnitTest.java [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+            assertEquals("HelloWorldThisisatest", result);
+        }
+    }
+
+    @Test
+    void givenInputStream_whenConvertingWithScannerFindAll_thenConvertInputStreamToStringStream() {
+        try (Scanner scanner = new Scanner(inputStream, StandardCharsets.UTF_8)) {
+            Stream<String> stringStream = scanner.findAll(".+")
+                    .map(MatchResult::group);
```
Comment (theangrydev) [inline core-java-modules/core-java-streams-6/src/test/java/com/baeldung/streams/inputstreamintostringstream/InputStreamToStringStreamUnitTest.java:36]: This went back to 8 spaces somehow. Are you sure you are using the Baeldung formatter?

**Review by theangrydev — APPROVED** [review]

**Pull request merged**